### PR TITLE
updates to allow a $kubernetes_install_method variable to accomplish:…

### PR DESCRIPTION
**Pull request by daveoy**
… https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#installing-kubeadm-kubelet-and-kubectl "installation without a package manager"

### CODE REVIEW

**Comment by chelnak** [comment]
Hello @daveoy ! Apologies for the lack of response and thank you for your contribution.

We will aim to review your contribution on the next Community Day. In the meantime, it looks like you have a couple of merge conflicts that need resolving. Could you get these sorted for us?

**Comment by daveoy** [comment]
> Hello @daveoy ! Apologies for the lack of response and thank you for your contribution.
> 
> We will aim to review your contribution on the next Community Day. In the meantime, it looks like you have a couple of merge conflicts that need resolving. Could you get these sorted for us?

thank you, that's done!

File: manifests/init.pp [hunk 1]
```diff
@@ -541,6 +541,16 @@
...
 #
 class kubernetes (
+  Enum['archive','package'] $kubernetes_install_method  = 'package',
+  Optional[String] $kubernetes_archive                  = "kube-bins-${kubernetes_version}.tar.gz",
+  Optional[String] $kubernetes_archive_checksum                  = undef,
+  Optional[String] $kubernetes_source                   = undef,
+  Optional[String] $crictl_archive                  = "crictl-v1.22.0-linux-amd64.tar.gz",
+  Optional[String] $crictl_archive_checksum         = undef,
+  Optional[String] $crictl_source                   = undef,
+  Optional[String] $cni_plugins_archive                  = "cni-plugins-linux-amd64-v0.8.2.tgz",
```
Comment (bastelfreak) [inline manifests/init.pp:551]: If you assign a value other than undef to the variable, the `Optional[]` datatype doesn't make any sense, you can replace it with `String[]`

File: files/kubelet.service [hunk 2]
```diff
@@ -0,0 +1,22 @@
+[Unit]
```
Comment (bastelfreak) [inline files/kubelet.service:1]: please add a comment like `# THIS FILE IS MANAGED BY PUPPET`

File: files/10-kubeadm.conf [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Note: This dropin only works with kubeadm and kubelet v1.11+
```
Comment (bastelfreak) [inline files/10-kubeadm.conf:1]: please add a comment like `# THIS FILE IS MANAGED BY PUPPET`

File: files/10-kubeadm.conf [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+[Service]
+Environment="KUBELET_KUBECONFIG_ARGS=--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf"
+Environment="KUBELET_CONFIG_ARGS=--config=/var/lib/kubelet/config.yaml"
+# This is a file that "kubeadm init" and "kubeadm join" generates at runtime, populating the KUBELET_KUBEADM_ARGS variable dynamically
+EnvironmentFile=-/var/lib/kubelet/kubeadm-flags.env
+# This is a file that the user can use for overrides of the kubelet args as a last resort. Preferably, the user should use
+# the .NodeRegistration.KubeletExtraArgs object in the configuration files instead. KUBELET_EXTRA_ARGS should be sourced from this file.
+EnvironmentFile=-/etc/default/kubelet
+ExecStart=
+ExecStart=/usr/bin/kubelet $KUBELET_KUBECONFIG_ARGS $KUBELET_CONFIG_ARGS $KUBELET_KUBEADM_ARGS $KUBELET_EXTRA_ARGS
```
Comment (bastelfreak) [inline files/10-kubeadm.conf:11]: please add the missing \n at the end of the line

File: manifests/init.pp [hunk 5]
```diff
@@ -541,6 +541,16 @@
 #
 #
 class kubernetes (
+  Enum['archive','package'] $kubernetes_install_method  = 'package',
+  Optional[String] $kubernetes_archive                  = "kube-bins-${kubernetes_version}.tar.gz",
```
Comment (bastelfreak) [inline manifests/init.pp:545]: If an empty string doesn't make sense, please switch to `String[1]`

File: manifests/init.pp [hunk 6]
```diff
@@ -541,6 +541,16 @@
...
+  Optional[String] $kubernetes_archive_checksum                  = undef,
+  Optional[String] $kubernetes_source                   = undef,
```
Comment (bastelfreak) [inline manifests/init.pp:547]: please fix the `=` alignment

File: manifests/init.pp (hunk 1 above)
Comment (daveoy) [reply manifests/init.pp]: examples have been moved to comment, should the appear in the docstrings above inside init.pp?

File: files/kubelet.service (hunk 2 above)
Comment (daveoy) [reply files/kubelet.service:2]: done

File: files/10-kubeadm.conf (hunk 3 above)
Comment (daveoy) [reply files/10-kubeadm.conf:2]: done

File: files/10-kubeadm.conf (hunk 4 above)
Comment (daveoy) [reply files/10-kubeadm.conf]: done

File: manifests/init.pp (hunk 5 above)
Comment (daveoy) [reply manifests/init.pp]: done

File: manifests/init.pp (hunk 6 above)
Comment (daveoy) [reply manifests/init.pp]: done

**Comment by daveoy** [comment]
@chelnak @bastelfreak Can we get this over the line?

**Comment by daveoy** [comment]
hi @chelnak @bastelfreak just noticed this is still open? can we please get this done?

**Comment by GSPatton** [comment]
Hello @daveoy 

Thank you for this PR. While we appreciate your contribution the team have decided that we aren’t going to move forward with this PR right now. We will add this to our list as a feature request and will make every attempt to consider it in the future.

Once again, thank you!

**Comment by daveoy** [comment]
Okay, just to add I'm using this on 10+ clusters in an air-gapped environment in production right now. I have a couple of updates to add to this PR as well, I'll push them to my fork and continue to reference it in case anyone ends up here looking for a solution to this problem.

**Pull request closed without merging**